ibis/backends/pandas/__init__.py

```diff
@@ -120,11 +124,30 @@ def compile(self, expr, *args, **kwargs):
         return expr
 
     def create_table(
-        self, table_name: str, obj=None, schema: sch.Schema | None = None
+        self,
+        name: str,
+        obj: pd.DataFrame | ir.Table | None = None,
+        *,
+        schema: sch.Schema | None = None,
+        database: str | None = None,
+        temp: bool | None = None,
+        overwrite: bool = False,
     ) -> ir.Table:
         """Create a table."""
+        if temp:
+            raise com.IbisInputError(
+                "Passing `temp=True` to the Pandas backend create_table method has no "
+                "effect: all tables are in memory and temporary. Avoid passing the "
+                "`temp` argument."
+            )
+        if database:
+            raise com.IbisInputError(
+                "Passing `database` to the Pandas backend create_table method has no "
+                "effect: Pandas cannot set a database. Avoid passing the `database` "
+                "argument."
+            )
```

Comment:
To support writing backend-agnostic code, perhaps these should be warnings (or just silently ignored)? That's easy to loosen the restrictions later though, so a decision on this now doesn't need to be a blocker.

Reply:
it's a bit sticky for me.
if someone passes an invalid argument (and they only use a single backend) I'd expect an error, since users ignore warnings (in my experience).
But we don't want to make it harder to write backend-agnostic code, so I guess warnings might be the way to go.

Reply:
made them warnings.

Reply:
Gah, sorry for the back-and-forth on this here. I guess there's a lot of backends that don't truly support these options (dask, polars, pyspark?, ...). We should either warn uniformly, or just revert back to erroring here for now. I don't feel strongly either way, but it's easier to relax restrictions later than to add them back.

Reply:
we'll need to distinguish eventually between `WillNeverBeImplemented` because, e.g. `temp` doesn't mean anything in pandas-land vs `NotImplemented` because `ibis` doesn't yet support a thing the backend can do.

My inclination is to `warn` if an option is (effectively) meaningless and `error` if it has meaning but isn't yet supported.

So I think that would translate to warning for pandas, dask, polars and maybe pyspark?

Reply:
I think that split makes sense to me.

Reply:
For PySpark, I think it's probably an error in both database case and temp view case. I think whether a view is permanent depends on whether there's a hive metastore in the mix or not.

I guess I'm +1 on erroring here so that people don't get the wrong idea about backend compatibility with DDL.

warnings indicate a meaningful difference in behavior IMO should be errors

Reply:
Ok -- how about `com.IbisError` for those backends where `temp` doesn't make sense and `NotImplementedError` otherwise?

Reply:
SGTM.

Reply:
That makes sense I think. Nice distinction between "will never be supported" and "may be eventually supported".

Reply:
Great conversation, thanks all!